staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -169,7 +180,15 @@ func (f *SimpleRestOptionsFactory) GetRESTOptions(resource schema.GroupResource)
 		ResourcePrefix:          resource.Group + "/" + resource.Resource,
 	}
 	if f.Options.EnableWatchCache {
-		ret.Decorator = genericregistry.StorageWithCacher(f.Options.DefaultWatchCacheSize)
+		sizes, err := ParseWatchCacheSizes(f.Options.WatchCacheSizes)
```

Comment:
Isn't this effectively switching off the changes from: cmd/kube-apiserver/app/server.go (to use the heuristic)?

Reply:
No, the code in the app/server.go gets applied first to update f.Options.WatchCacheSizes